### DISCUSSION TITLE
突然提示版本过低

### DISCUSSION THREAD

**Discussion by codez-yydd** [simple-robot/Discussions#75] [Q&A] [answered]
昨天还用的好好的，一过12点提示QQ版本过低，需要升级，这个怎么解决？

**Comment by ForteScarlet** [comment, marked as answer]
更新[mirai组件版本](https://github.com/simple-robot/simbot-component-mirai/releases)到[最新版本](https://github.com/simple-robot/simbot-component-mirai/releases/latest)并参考各版本更新日志

**Reply by codez-yydd** [reply]
<dependency>
			<groupId>love.forte.simple-robot</groupId>
			<artifactId>component-mirai</artifactId>
			<version>2.5.2</version>
		</dependency>
这个就是最新的呀

**Reply by ForteScarlet** [reply]
如果使用 2.x，则尝试删除 `device.json` 、`cache` 等文件/目录后再次尝试。
其他参考：[discussions#69](https://github.com/orgs/simple-robot/discussions/69)

建议尽早重构到3.x，2.x不会一直维护下去

**Reply by codez-yydd** [reply]
好的谢谢，我已经在转换3.0 m5

**Reply by codez-yydd** [reply]
你好，能直接发给我版本依赖吗，我在版本依赖这里卡了很久了，simboot-core-spring-boot-starter 3.0.0-RC.3没有@EnableSimbot注解吗？
当我降低版本为3.0.0-RC有这个注解了，但是启动还是提示我code=235, title=温馨提示, message=当前QQ版本过低，请升级至最新版本后再登录。点击进入下载页面
simbot-component-mirai-core 为3.0.0.0-M5
昨天我刚刚了解你的组件，2.5.2昨天已经开发完成并且上线在运行，刚过12点就没了，救救孩子吧，我看api文档大多都要重新再去修改，救救孩子

**Reply by ForteScarlet** [reply]
v2跟v3 **完全不同**，重构是必然的，可以参考[手册](http://simbot.forte.love/) 中的 _快速开始_ 等相关文档。

v3中这类问题也可以尝试尝试删除 `device.json`、`cache` 目录等操作，以及修改协议为其他类型（例如 `ANDROID_PAD`，参考 [mirai组件手册-bot配置](https://component-mirai.simbot.forte.love/docs/bot-config) 中的 `config.protocol` ）

`code=235` `code=237` 等问题依旧参考 #69 以及其中的其他参考，比如尝试更换其他账号。
这是tx风控的一种体现，没有完美的解决方案。

其他参考：https://github.com/mamoe/mirai/issues/2492

> 没有@EnableSimbot注解

此问题参考 #70

**Reply by codez-yydd** [reply]
我从2已经重构到3，两个账号都是只能使用不到半天时间就提示版本低，不能登录了，我的机器人是对于我的用户作为网站签到使用的，并没有什么敏感问题，这种是tx风险没有办法解决吗？

**Reply by ForteScarlet** [reply]
风控的规则是tx定的，它**可能**包含很多方面，比如频繁登录，举报次数甚至于账号等级等方面（以上均为个人猜想，无根据），这没法预测也没有100％有效解决方案。

如果还出现出问题，清理`device.json` `cache` 等后再次尝试。

你可以尝试为账号开启设备保护、充值会员、挂机养号增加等操作。但是这些都是**尝试**，只是我的个人猜测，不保证效果

**Reply by codez-yydd** [reply]
好的，谢谢，项目很nice

**Reply by ForteScarlet** [reply]
感谢支持

**Reply by codez-yydd** [reply]
我更换账号也不能使用了，之前提示过的正常电脑也不能登陆，同一台登陆其他没问题，登陆用过就是版本低
 code=45, title=禁止登录, message= 登录失败，建议升级最新版本后重试，或通过问题反馈与我们联系。, errorInfo=)

**Reply by ForteScarlet** [reply]
保证版本更新的同时尝试清理 `device.json` `cache` 等后再次尝试，或者尝试调整bot协议类型为其他类型。
风控随时都可能会被升级，可以前往 [mirai的issues](https://github.com/mamoe/mirai/issues) 处翻阅于此相关的最新讨论，例如 https://github.com/mamoe/mirai/issues/2453 、https://github.com/mamoe/mirai/issues/2492 等。

**Reply by codez-yydd** [reply]
"config": {
    "protocol": "ANDROID_WATCH"
  }
修改协议之后，之前风险的账号可以成功登陆使用了，谢谢
